cpp/tests/sort/stable_sort_tests.cpp

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_utilities.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/table_utilities.hpp>
+#include <cudf_test/type_lists.hpp>
+
+#include <cudf/column/column_factories.hpp>
+#include <cudf/copying.hpp>
+#include <cudf/fixed_point/fixed_point.hpp>
+#include <cudf/sorting.hpp>
+#include <cudf/table/table_view.hpp>
+#include <cudf/types.hpp>
+#include <cudf/utilities/type_dispatcher.hpp>
+
+#include <vector>
+
+namespace cudf {
+namespace test {
+void run_stable_sort_test(table_view input,
+                          column_view expected_sorted_indices,
+                          std::vector<order> column_order         = {},
+                          std::vector<null_order> null_precedence = {})
+{
+  auto got_sort_by_key_table      = sort_by_key(input, input, column_order, null_precedence);
+  auto expected_sort_by_key_table = gather(input, expected_sorted_indices);
+
+  CUDF_TEST_EXPECT_TABLES_EQUAL(expected_sort_by_key_table->view(), got_sort_by_key_table->view());
+}
+
+using TestTypes = cudf::test::Concat<cudf::test::IntegralTypesNotBool,
+                                     cudf::test::FloatingPointTypes,
+                                     cudf::test::DurationTypes,
+                                     cudf::test::TimestampTypes>;
```

Comment:
The test types here do not include `bool` but the test code below includes special logic for handling `bool` types.

```suggestion
using TestTypes = cudf::test::Concat<cudf::test::NumericTypes,  // includes integers, floats, and bool
                                     cudf::test::ChronoTypes>; // includes timestamp and duration types
```

Are there test cases for `FixedPointTypes`? Maybe those are not sortable?

Reply:
Fixed point tests added.